packages/arb-validator-core/ethbridgetest/message_test.go

```diff
@@ -30,230 +41,353 @@ import (
 	"github.com/offchainlabs/arbitrum/packages/arb-validator-core/test"
 )
 
-func init() {
+var privateKeyHex = "27e926925fb5903ee038c894d9880f74d3dd6518e23ab5e5651de93327c7dffa"
+
+var auth *bind.TransactOpts
+var tester *messagetester.MessageTester
+var client *ethclient.Client
+
+var addr1 common.Address
+var addr2 common.Address
+var addr3 common.Address
+
+var errHash = errors.New("ethbridge calculated wrong hash")
+var errMsgHash = errors.New("ethbridge calculated wrong message hash")
+
+func TestMain(m *testing.M) {
+	addr1[0] = 76
+	addr1[19] = 93
+	addr2[0] = 43
+	addr2[19] = 12
+	addr3[0] = 73
+	addr3[19] = 85
+
+	var err error
 
+	auth, err = test.SetupAuth(privateKeyHex)
+	if err != nil {
+		log.Fatal(err)
+	}
+	client, err = ethclient.Dial(test.GetEthUrl())
+	if err != nil {
+		log.Fatal(err)
+	}
+	_, tx, deployedTester, err := messagetester.DeployMessageTester(auth, client)
+	if err != nil {
+		log.Fatal(err)
+	}
+	_, err = ethbridge.WaitForReceiptWithResults(context.Background(), client, auth.From, tx, "DeployMessageTester")
+	if err != nil {
+		log.Fatal(err)
+	}
+	tester = deployedTester
+	code := m.Run()
+	os.Exit(code)
 }
 
-func TestMessageHashing(t *testing.T) {
-	auth, err := test.SetupAuth("27e926925fb5903ee038c894d9880f74d3dd6518e23ab5e5651de93327c7dffa")
+func TestTransactionMessage(t *testing.T) {
+	msg := message.DeliveredTransaction{
+		Transaction: message.Transaction{
+			Chain:       addr3,
+			To:          addr1,
+			From:        addr2,
+			SequenceNum: big.NewInt(74563),
+			Value:       big.NewInt(89735406),
+			Data:        []byte{65, 23, 68, 87, 12},
+		},
+		BlockNum:  common.NewTimeBlocks(big.NewInt(87962345)),
+		Timestamp: big.NewInt(35463245),
+	}
+	bridgeHash, err := tester.TransactionHash(
+		nil,
+		msg.Chain.ToEthAddress(),
+		msg.To.ToEthAddress(),
+		msg.From.ToEthAddress(),
+		msg.SequenceNum,
+		msg.Value,
+		msg.Data,
+		msg.BlockNum.AsInt(),
+		msg.Timestamp,
+	)
 	if err != nil {
 		t.Fatal(err)
 	}
-	client, err := ethclient.Dial(test.GetEthUrl())
+	if bridgeHash != msg.CommitmentHash().ToEthHash() {
+		t.Error(errHash)
+	}
+
+	messageBridgeHash, err := tester.TransactionMessageHash(
+		nil,
+		msg.Chain.ToEthAddress(),
+		msg.To.ToEthAddress(),
+		msg.From.ToEthAddress(),
+		msg.SequenceNum,
+		msg.Value,
+		msg.Data,
+		msg.BlockNum.AsInt(),
+		msg.Timestamp,
+	)
 	if err != nil {
 		t.Fatal(err)
 	}
-	_, tx, tester, err := messagetester.DeployMessageTester(auth, client)
+
+	if messageBridgeHash != message.DeliveredValue(msg).Hash().ToEthHash() {
+		t.Error(errMsgHash)
+	}
+}
+
+func TestTransactionBatchMessage(t *testing.T) {
+	chain := addr3
+	data := []byte{65, 23, 68, 87, 12}
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [avoid magic numbers like '12', create a named constant for it](https://app.codacy.com/gh/OffchainLabs/arbitrum/pullRequest?prid=5428340)